packages/framer-motion/src/projection/node/create-projection-node.ts

```diff
@@ -108,21 +108,25 @@ function resetDistortingTransform(
     }
 }
 
-function isOptimisedAppearTree(projectionNode: IProjectionNode) {
+function isOptimisedTransformAnimationInTree(projectionNode: IProjectionNode) {
     projectionNode.hasCheckedOptimisedAppear = true
     if (projectionNode.root === projectionNode) return false
 
     const { visualElement } = projectionNode.options
 
     if (!visualElement) {
         return false
-    } else if (getOptimisedAppearId(visualElement)) {
+    } else if (
+        window.MotionHasOptimisedTransformAnimation!(
+            getOptimisedAppearId(visualElement)
+        )
```

Comment:
Is it safer to do 
```suggestion
        window.MotionHasOptimisedTransformAnimation?.(
            getOptimisedAppearId(visualElement)
        )
```

Reply:
So this function is wrapped in a check for these functions - `?.` is safer but in this case I think it's ok to avoid the overhead (?. gets transpiled to quite a noisy series of checks and we want to quickly look up a tree here)

Reply:
> So this function is wrapped in a check for these functions 

ah cool, missed that bit.